mkosi/__init__.py

```diff
@@ -3721,20 +3825,69 @@ def make_tar(args: CommandLineArguments, root: str, do_run_build_script: bool, f
     if for_cache:
         return None
 
-    tar_root_dir = f"{root}/usr" if args.usr_only else root
+    root_dir = f"{root}/usr" if args.usr_only else root
 
     with complete_step("Creating archive"):
         f: BinaryIO = cast(BinaryIO, tempfile.NamedTemporaryFile(dir=os.path.dirname(args.output), prefix=".mkosi-"))
         # OpenMandriva defaults to bsdtar(libarchive) which uses POSIX argument list so let's keep a separate list
         if shutil.which("bsdtar") and args.distribution == Distribution.openmandriva:
-            _tar_cmd = ["bsdtar", "-C", tar_root_dir, "-c", "-J", "--xattrs", "-f", "-", "."]
+            cmd = ["bsdtar", "-C", root_dir, "-c", "--xattrs", "-f", "-", "."]
         else:
-            _tar_cmd = ["tar", "-C", tar_root_dir, "-c", "-J", "--xattrs", "--xattrs-include=*"]
+            cmd = ["tar", "-C", root_dir, "-c", "--xattrs", "--xattrs-include=*"]
             if args.tar_strip_selinux_context:
-                _tar_cmd.append("--xattrs-exclude=security.selinux")
-            _tar_cmd.append(".")
+                cmd.append("--xattrs-exclude=security.selinux")
+            cmd.append(".")
 
-        run(_tar_cmd, env={"XZ_OPT": "-T0"}, stdout=f)
+        compressor = compressor_command(should_compress_output(args))
+
+        with spawn(cmd, stdout=subprocess.PIPE) as tar:
+            run(compressor, stdin=tar.stdout, stdout=f, delay_interrupt=False)
```

Comment:
Is there an advantage not to use tar's `--use-compress-program` as `f"--use-compress-program='{compressor}'"`? The short option is different between GNU tar and BSD tar, but the long option has the same name.

Reply:
Indeed. I'll rework it like that.